pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -359,6 +359,16 @@ public void addSegment(String segmentName, TableConfig tableConfig, IndexLoading
     _serverMetrics.addValueToTableGauge(_tableNameWithType, ServerGauge.SEGMENT_COUNT, 1L);
   }
 
+  /*
+   * This method is implemented to allow refreshing the segments in realtime tables.
+   */
+  @Override
+  public void addSegment(File indexDir, IndexLoadingConfig indexLoadingConfig)
```

Comment:
Move this implementation to the `BaseTableDataManager`. It is common for both offline and realtime table